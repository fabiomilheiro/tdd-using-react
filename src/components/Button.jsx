import React, { Component } from "react";
import "./Button.scss";

class Button extends Component {
  render() {
    return (
      <button
        id={this.props.id}
        onClick={this.props.onClick}
        value={this.props.text}
      >
        {this.props.text || "All locations"}
      </button>
    );
  }
}

export default Button;
