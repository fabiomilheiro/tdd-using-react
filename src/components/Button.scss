button {
  font-size: 18px;
  margin: 12px 6px;
  padding: 6px;
}
