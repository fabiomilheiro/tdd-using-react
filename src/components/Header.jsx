import React, { Component } from "react";

class Header extends Component {
  render() {
    return (
      <header>
        <img src="/images/logo.png" alt="" />
      </header>
    );
  }
}

export default Header;
