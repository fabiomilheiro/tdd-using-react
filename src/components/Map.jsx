import React, { Component } from "react";
import "./Map.scss";

class Map extends Component {
  render() {
    return (
      <div className="map">
        <img src={this.getImageSource()} alt="" />
      </div>
    );
  }

  getImageSource = () => {
    return `/images/${this.props.location || "None"}.png`;
  };
}

export default Map;
