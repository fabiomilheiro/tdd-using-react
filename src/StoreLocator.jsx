import React, { Component } from "react";
import Header from "./components/Header";
import Button from "./components/Button";
import Map from "./components/Map";
import "./store-locator.scss";
import Axios from "axios";

class StoreLocator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedLocation: "",
      locations: [{ location: "x" }, { location: "y" }, { location: "w" }],
    };
  }

  async componentDidMount() {
    const locationsResponse = await Axios.get("/data/locations.json");
    this.setState({
      locations: locationsResponse.data,
    });
  }
  render() {
    return (
      <div className="store-locator">
        <Header />
        {this.state.locations.map((l, id) => (
          <Button
            key={id}
            id={l.location}
            text={l.location}
            onClick={this.chooseMap}
          />
        ))}
        <Map location={this.state.selectedLocation} />
      </div>
    );
  }

  chooseMap = (event) => {
    this.setState({
      selectedLocation: event.target.value,
    });
  };
}

export default StoreLocator;
