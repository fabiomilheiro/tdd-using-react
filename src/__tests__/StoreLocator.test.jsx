import React from "react";
// import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import StoreLocator from "../StoreLocator";
import Header from "./../components/Header";
import Button from "./../components/Button";
import Map from "./../components/Map";
import Axios from "axios";

const locationsResponse = {
  data: [
    {
      location: "Test Portland",
    },
    {
      location: "Test Astoria",
    },
    {
      location: "",
    },
  ],
};

describe("Store locator", () => {
  let storeLocator;
  beforeEach(async () => {
    Axios.get.mockImplementation((url) => {
      if (url === "/data/locations.json") {
        return Promise.resolve(locationsResponse);
      }

      return Promise.reject("");
    });
    storeLocator = shallow(<StoreLocator />);
    await storeLocator.instance().componentDidMount();
  });

  test("renders shallow without crashing", () => {});

  test("Renders header", () => {
    const headers = storeLocator.find(Header);
    expect(headers.length).toBe(1);
  });

  test("Renders buttons for all shops + 1", () => {
    const buttons = storeLocator.find(Button);
    expect(buttons.length).toBe(3);
  });

  test("Renders one map", () => {
    const maps = storeLocator.find(Map);
    expect(maps.length).toBe(1);
  });
});

describe("Choose map", () => {
  let storeLocator;
  beforeEach(async () => {
    storeLocator = shallow(<StoreLocator />);
    await storeLocator.instance().componentDidMount();
  });

  test("Updates state using the location passed to it", () => {
    var event = { target: { value: "test" } };
    storeLocator.instance().chooseMap(event);
    expect(storeLocator.instance().state.selectedLocation).toBe("test");
  });
});
