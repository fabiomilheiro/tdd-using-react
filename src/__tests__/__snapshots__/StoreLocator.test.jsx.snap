// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store locator renders 1`] = `
<div
  className="store-locator"
>
  <header>
    <img
      alt=""
      src="/images/logo.png"
    />
  </header>
  <button
    id="x"
    value="x"
  >
    x
  </button>
  <button
    id="y"
    value="y"
  >
    y
  </button>
  <button
    id="w"
    value="w"
  >
    w
  </button>
  <div
    className="map"
  >
    <img
      alt=""
      src="/images/None.png"
    />
  </div>
</div>
`;
